### PR TITLE
drivers: ieee802154_nrf5: Block on net_pkt allocation in the RX path

### PR DESCRIPTION
**Pull request by rlubos**
Currently, if no net_pkt's are avaliable, the radio driver RX thread
drops the 802.15.4 frame silently. This causes undesired behaviour,
where we can drop the packet which has already been acknowledged at
the 802.15.4 level.

Fix this, by blocking the RX thread if no net_pkt is avaliable. The
packets received while the RX thread is blocked will be accumulated in
the underlying nRF 802.15.4 driver, and eventually when it runs out of
internal buffers before the thread is unblocked, it'll stop
acknowledging the incoming frames.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

**Comment by rlubos** [comment]
CC @MarekPorwisz

**Review by MarekPorwisz — APPROVED** [review]

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 1]
```diff
@@ -134,7 +134,7 @@ static void nrf5_rx_thread(void *arg1, void *arg2, void *arg3)
 		LOG_DBG("Frame received");
 
 		pkt = net_pkt_rx_alloc_with_buffer(nrf5_radio->iface, pkt_len,
-						   AF_UNSPEC, 0, K_NO_WAIT);
+						   AF_UNSPEC, 0, K_FOREVER);
 		if (!pkt) {
 			LOG_ERR("No pkt available");
 			goto drop;
```
Comment (jukkar) [inline drivers/ieee802154/ieee802154_nrf5.c:140]: The check becomes unnecessary if we wait forever.

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 2]
```diff
@@ -134,7 +134,7 @@ static void nrf5_rx_thread(void *arg1, void *arg2, void *arg3)
 		LOG_DBG("Frame received");
 
 		pkt = net_pkt_rx_alloc_with_buffer(nrf5_radio->iface, pkt_len,
```
Comment (jukkar) [inline drivers/ieee802154/ieee802154_nrf5.c:136]: Could you add a comment telling why we can wait forever here, I know it is in commit message but it helps if there is description here too.

**Review by jukkar — CHANGES_REQUESTED** [review]

**Comment by rlubos** [comment]
@jukkar Thanks, comments addressed.

**Review by jukkar — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]
Good one! Thanks :+1:

**Review by tbursztyka — APPROVED** [review]
in your commit message: s/avaliable/available

**Pull request merged**